test/test_scandir.py

```diff
@@ -250,6 +250,23 @@ def test_unicode(self):
         self.assertEqual(entry.name, 'file1.txt')
         self.assertEqual(entry.path, os.path.join(path, 'file1.txt'))
 
+    def test_os_walk_unicode_handling(self):
```

Comment:
Why "os_walk"? It's a function of scandir.walk(). Just remove "os_" from the test name.